keel-scheduler/src/main/kotlin/com/netflix/spinnaker/keel/scheduler/handler/MonitorOrchestrationsHandler.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.keel.scheduler.handler
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.keel.IntentActivityRepository
+import com.netflix.spinnaker.keel.orca.OrcaExecutionStatus
+import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.scheduler.MonitorOrchestrations
+import com.netflix.spinnaker.q.MessageHandler
+import com.netflix.spinnaker.q.Queue
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import java.time.Duration
+
+
+/**
+ * This handler follows up on the launched orchestrations, and records
+ * success and failure by kind.
+ */
+
+@Component
+class MonitorOrchestrationsHandler
+@Autowired constructor(
+  override val queue: Queue,
+  private val intentActivityRepository: IntentActivityRepository,
+  private val orcaService: OrcaService,
+  private val registry: Registry
+) : MessageHandler<MonitorOrchestrations> {
+
+  private val backoffMs = Duration.ofMillis(10000)
+
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  private val orchestrationsStatusId = registry.createId("intent.orchestrations.status")
+
+  override fun handle(message: MonitorOrchestrations) {
+    val orchestrationIds = intentActivityRepository.getCurrent(message.intentId)
+    orchestrationIds.forEach {
+      val task = orcaService.getTask(it)
+      val orcaExecutionStatus = parseStatus(task)
+
+      when (orcaExecutionStatus) {
+        OrcaExecutionStatus.NOT_STARTED, OrcaExecutionStatus.RUNNING -> {
+          log.debug("orchestration $it for intent ${message.intentId} is RUNNING")
```

Comment:
Use structured logs where possible (and consistently). For example:

```
log.debug("orchestration for intent is running (intentId: {}, taskId: {})", value("intentId", message.intentId), value("taskId", it))
```

etc... This will make things more searchable.

Reply:
Yes I forgot, good reminder!